packages/web/components/MegaMenu/MegaMenuHeader.tsx

```diff
@@ -52,6 +80,307 @@ const Logo: React.FC<LogoProps> = ({ link, ...props }) => {
+      .pipe(
+        filter((searchValue: string) => {
+          if (searchValue.length >= 1) return true;
+
+          resetResults();
+          return false;
+        }),
+        debounceTime(300),
+        distinctUntilChanged(),
+        switchMap((queryString) =>
+          forkJoin([
+            from(searchPlayers(queryString)),
+            from(searchGuilds(queryString)),
+          ]),
+        ),
+        shareReplay(1),
+      )
+      .subscribe(([{ players: p }, { guilds: g }]) => {
+        setSearchResults({ players: p, guilds: g });
+      });
+    return () => searchSubscription?.unsubscribe();
+  }, []);
+
+  const isBodyEmpty = players.length + guilds.length === 0;
+  return (
+    <Modal {...{ isOpen }} {...{ onClose }} scrollBehavior="inside">
+      <ModalOverlay />
+      <ModalContent
+        overflow="hidden"
+        top="max(4rem, 8vh)"
+        shadow="lg"
+        maxH="700px"
+        maxW="500px"
+        aria-expanded="true"
+        marginTop={1}
+        p={0}
+      >
+        <Flex
+          flexDirection="column"
+          alignItems="center"
+          minWidth={40}
+          pos="relative"
+          align="stretch"
+          ref={dropdown}
+        >
+          <Box as="form" onSubmit={handleSubmit} w="full" color="white">
+            <InputGroup
+              justifyContent="flex-start"
+              h="fit-content"
+              p={2}
+              my="auto"
+              bg={{
+                base: '#FFFFFF05',
+              }}
+              border={{ base: '1px solid #2B2244' }}
+              borderRadius={4}
+            >
+              <InputLeftElement
+                pointerEvents="none"
+                children={
+                  <Image src={SearchIcon} alt="search" height={4} width={4} />
+                }
+              />
+              <Input
+                variant="unstyled"
+                color="white"
+                w="100%"
+                placeholder="Find Players or Guilds"
+                _placeholder={{ color: 'whiteAlpha.500' }}
+                value={query}
+                onChange={({ target: { value } }) => setQuery(value)}
+                size="sm"
+                fontSize="md"
+                ref={searchBarRef}
+              />
+            </InputGroup>
+          </Box>
+          <ModalBody
+            sx={{
+              '&::-webkit-scrollbar': {
+                width: '5px',
+              },
+              bg: !isBodyEmpty ? 'white' : 'transparent',
+            }}
+            w="100%"
+            maxH="66vh"
+            p={0}
+          >
+            {!isBodyEmpty && (
+              <Box
+                w="100%"
+                borderRadius="0.25rem"
+                css={{
+                  transform: 'translate3d(0px, 15px, 0px)',
+                }}
+              >
+                <Box as="ul" role="listbox" pb={8}>
+                  {players.length > 0 && <ResultsTitle>Players</ResultsTitle>}
```

Comment:
This will create a: `<ul><p/><li/>…</ul>` structure &, by rights, the `<p/>` should be outside the `<ul/>`.
```suggestion
                {players.length > 0 && <ResultsTitle>Players</ResultsTitle>}
                <Box as="ul" role="listbox" pb={8}>
```
Or, even better yet, don't create the `<ul/>` if there are no players:
```suggestion
                {players.length > 0 && (
                  <>
                    <ResultsTitle>Players</ResultsTitle>
                    <Box as="ul" role="listbox" pb={8}>
                      ⋮
                    </Box>
                  </>
                )}
```